### PR TITLE
Add library for uploading results to Cloud Storage

### PR DESCRIPTION
**Pull request by bochunz**

### CODE REVIEW

File: storage/storage.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	"golang.org/x/net/context"
+)
+
+// DatastoreClient is a struct for interacting with Google Cloud Datastore
+type DatastoreClient struct {
+	client *datastore.Client
+}
+
+// IstioPerfResult is the data structure for uploading a test run result to Google Cloud Datastore
+type IstioPerfResult struct {
```
Comment (ldemailly) [inline storage/storage.go:37]: Can we use the (Http)RunnerResults directly somehow is there a default mapping where the datastore just stores every exported field (like json does) ?

File: storage/storage.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+// (no std output by default)
+
+package storage // import "istio.io/fortio/storage"
+
+import (
+	"time"
+
+	"istio.io/fortio/log"
+
+	"cloud.google.com/go/datastore"
```
Comment (ldemailly) [inline storage/storage.go:27]: so far fortio has no external dependencies outside of grpc (and the grpc part is optional)

I wonder if we should put that gcp specific part somewhere else (istio/istio/tools/perf or istio/test-infra...)

or is there a way to just POST json directly?

Comment (bochunz) [reply storage/storage.go:27]: There is. But will be ugly in code (we need to somehow manually grab credentials, and manually construct a giant JSON).

Comment (bochunz) [reply storage/storage.go:27]: Can be done somewhat easier with a shell + python script though.

File: storage/storage.go (hunk 1 above)
Comment (bochunz) [reply storage/storage.go:37]: I'd expect what we upload to be more or less different from what those internal data structures. The ResultJSON here should probably just be the serialization of RunnerResults. And we only put fields we explicitly want to index here.

**Comment by ldemailly** [comment]
closing in favor of tsv export

**Pull request closed without merging**